### DISCUSSION TITLE
What to do with places widget

### DISCUSSION THREAD

**Discussion by Haroenv** [algolia/instantsearch#4904] [Ideas]
Algolia places will no longer exist soon, but the use case for synchronising location and algolia parameters still seems interesting. 

In geosearch we only support a bounding box, and the existing places widget only supports aroundLatLng, I wonder if those two paradigms shouldn't exist at the same time?

These are the options I see:

1. remove places widget without replacement
2. add a connector responsible for aroundLatLng and "location query" (external to algolia parameters) (what places widget does now)
1. add LatLng and location query functionality to the geo search connector
4. writing a fully new "geo" connector that can handle both suggestion of locations, and bounding box around a map view
3. add something of those options, but only as a guide

If we do nothing, we are essentially doing option 1, but people could still take the current code of the places widget and adapt that to different providers

references: #4775

**Comment by placesapi** [comment]
Hi, we use InstantSearch a lot and faced similar problems with out clients so we created [PlacesAPI](https://www.placesapi.co.uk). It's an Algolia Places replacement (also built on Algolia) and it's a breeze to swap out the native Places widget with ours, see Demo 6 on our site.
